classes/Integration/Form/FormidableForms.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/************************************
+ * Copyright (c) 2020, Popup Maker
+ ************************************/
+
+/**
+ * Handles the integration with Formidable Forms (https://wordpress.org/plugins/formidable/)
+ *
+ * @since 1.12
+ */
+class PUM_Integration_Form_FormidableForms extends PUM_Abstract_Integration_Form {
+
+	/**
+	 * Unique key identifier for this provider.
+	 *
+	 * @var string
+	 */
+	public $key = 'formidableforms';
+
+	/**
+	 * Only used to hook in a custom action for non AJAX based submissions.
+	 *
+	 * Could be used for other initiations as well where needed.
+	 */
+	public function __construct() {
+		add_action( 'frm_after_create_entry', array( $this, 'on_success' ), 1, 2 );
+	}
+
+	/**
+	 * Text label that will be used throughout the various options screens.
+	 *
+	 * @return string
+	 */
+	public function label() {
+		return __( 'Formidable Forms' );
+	}
+
+	/**
+	 * Should return true when the required form plugin is active.
+	 *
+	 * @return bool
+	 */
+	public function enabled() {
+		return class_exists( 'FrmEntry' );
+	}
+
+	/**
+	 * Return a useable array of all forms from this provider.
+	 *
+	 * @return array
+	 */
+	public function get_forms() {
+		return FrmForm::getAll();
+	}
+
+	/**
+	 * Return a single form by ID.
+	 *
+	 * @param string $id The ID of the form.
+	 * @return mixed
+	 */
+	public function get_form( $id ) {
+		return FrmForm::getOne( intval( $id ) );
+	}
+
+	/**
+	 * Returns an array of options for a select list.
+	 * Should be in the format of $formId => $formLabel.
+	 *
+	 * @return array The array of options
+	 */
+	public function get_form_selectlist() {
+		$form_selectlist = [];
+
+		$forms = $this->get_forms();
+
+		foreach ( $forms as $form ) {
+			$form_selectlist[ $form->id ] = $form->name;
+		}
+
+		return $form_selectlist;
+	}
+
+	/**
+	 * Hooks in a success functions specific to this provider for non AJAX submission handling.
+	 *
+	 * @param int $entry_id The ID of the entry added.
+	 * @param int $form_id The ID of the form.
+	 */
+	public function on_success( $entry_id, $form_id ) {
+		// @see pum_integrated_form_submission
+		pum_integrated_form_submission(
```

Comment:
When using the on_success method via PHP, you need to confirm a few things:

Does the event you hooked into also get called if the form is submitted over AJAX?

If so should we prevent this from being called as it could have already been done via JS. Specifically I'm thinking tracking of conversions for example.


2. If it doesn'

Reply:
@danieliser So, the PHP hook gets fired both during PHP submission and AJAX submission. The free version only has the PHP submission so we will need to keep that. But, we need to determine a way to only fire that if the AJAX submission did not happen. Since they pass $form_id, I might be able to get the form's settings to see if AJAX submission is configured.

So, if the form has AJAX submission enabled, don't fire our form submission function in this method. I'll review the docs and see what I can find.

Reply:
Got this working in the recent commit!